spec/notifications-status-view-spec.js

```diff
@@ -0,0 +1,100 @@
+
+const {Notification} = require('atom');
+const {generateFakeFetchResponses, generateException} = require('./helper');
+
+describe("Notifications Count", () => {
+  let [workspaceElement, statusBarManager, notificationsCountContainer] = Array.from([]);
+
+  beforeEach(() => {
+    workspaceElement = atom.views.getView(atom.workspace);
+    atom.notifications.clear();
+
+    waitsForPromise(() =>
+      Promise.all([
+        atom.packages.activatePackage('notifications'),
+        atom.packages.activatePackage('status-bar')
+      ]));
+
+    runs(() => {
+      statusBarManager = atom.packages.getActivePackage('notifications').mainModule.statusBarManager;
+      notificationsCountContainer = workspaceElement.querySelector('.notifications-count');
+    });
+  });
+
+  describe("when the package is activated", () =>
+    it("attaches an .notifications-count element to the dom", () => {
+      expect(statusBarManager.count).toBe(0);
+      expect(notificationsCountContainer).toExist();
+    })
+  );
+
+  describe("when there are notifications before activation", () => {
+    beforeEach(() =>
+      waitsForPromise(() =>
+        // Wrapped in Promise.resolve so this test continues to work on earlier versions of Atom
```

Comment:
Good news: this isn't needed anymore :)

And if you want you can try converting this spec file to use async/await (see https://github.com/atom/archive-view/pull/57 for example), but it's not necessary.

I'll try to give an actual content review in the next few days.

Reply:
It looks like it only works with the [async-spec-helper.js](https://github.com/atom/archive-view/pull/57/files#diff-41ed07bdd040722c36a531c9e36219ff) file in the pr

did you want me to add that file?

It would be really nice if we could move to jasmine 2 with [atom-jasmine2-test-runner](https://www.npmjs.com/package/atom-jasmine2-test-runner)

That has built in support for async/await and more

Reply:
atom-jasmine2-test-runner even works with jasmine 1 specs so we wouldn't have to move them all over at once